.github/workflows/builder.yml

```
@@ -0,0 +1,313 @@
+      channel: ${{ steps.version.outputs.channel }}
+      publish: ${{ steps.version.outputs.publish }}
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        uses: actions/setup-python@v2.2.1
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+
+      - name: Get information
+        id: info
+        uses: home-assistant/actions/helpers/info@master
+
+      - name: Get version
+        id: version
+        uses: home-assistant/actions/helpers/version@master
+        with:
+          type: ${{ env.BUILD_TYPE }}
+
+      - name: Verify version
+        uses: home-assistant/actions/helpers/verify-version@master
+        with:
+          ignore-dev: true
+
+  build_python:
+    name: Build PyPi package
+    needs: init
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        if: needs.init.outputs.publish == "true"
+        uses: actions/checkout@v2
+
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        if: needs.init.outputs.publish == "true"
+        uses: actions/setup-python@v2.2.1
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+
+      - name: Build package
+        if: needs.init.outputs.publish == "true"
+        shell: bash
+        run: |
+          pip install twine wheel
+          python setup.py sdist bdist_wheel
+
+      - name: Upload package
+        if: needs.init.outputs.publish == "true"
+        shell: bash
+        run: |
+          export TWINE_USERNAME="$(twineUser)"
+          export TWINE_PASSWORD="$(twinePassword)"
+
+          twine upload dist/* --skip-existing
+
+  build_base:
+    name: Build ${{ matrix.arch }} base core image
+    needs: init
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: ${{ fromJson(needs.init.outputs.architectures) }}
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        if: needs.init.outputs.channel == "dev"
+        uses: actions/setup-python@v2.2.1
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+
+      - name: Adjust nightly version
+        if: needs.init.outputs.channel == "dev"
+        shell: bash
+        run: |
+          python3 -m pip install packaging
+          python3 -m pip install .
+          python3 script/version_bump.py nightly
+          version="$(python setup.py -V)"
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.GIT_USER }}
+          password: ${{ secrets.GIT_TOKEN }}
```

We don't need GitHub PAT's anymore.

```suggestion
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GITHUB_TOKEN }}
```